### DISCUSSION TITLE
Get only validated fields to array from DTO request

### DISCUSSION THREAD

**Discussion by julianphp** [spatie/laravel-data#371] [Q&A] [answered]
Hi, I have been looking but I have not found anything, is it not possible from a DTO from a request, to get only the fields that are in the dto/validation to array? because with $request0->all() is taking all the fields that are sent from the resquet and I only want the ones that have been really validated. Something like laravel has with the ->validated() method.

**Comment by drew-acheson** [comment, marked as answer]
The way I have been using the library this works exactly like it sounds like you want it to.  

```
    public function create(Request $request): JsonResource
    {
        $insert_data = UpsertObject::from($request);

        // This returns only the properties that are defined in the UpsertObject constructor and ignores others in the request.
        DD($insert_data->toArray());
        ....
    }
```

Perhaps you can provide some code to show what you're doing?

**Reply by julianphp** [reply]
> The way I have been using the library this works exactly like it sounds like you want it to.
> 
> ```
>     public function create(Request $request): JsonResource
>     {
>         $insert_data = UpsertObject::from($request);
> 
>         // This returns only the properties that are defined in the UpsertObject constructor and ignores others in the request.
>         DD($insert_data->toArray());
>         ....
>     }
> ```
> 
> Perhaps you can provide some code to show what you're doing?

The truth is that I don't know what happened when I saw that it was returning parameters that were not in the DTO, but I have checked again in several ways and indeed it doesn't happen, it only returns the defined parameters.
